cloudflare/resource_cloudflare_zone_settings_override.go

```diff
@@ -0,0 +1,665 @@
+package cloudflare
+
+import (
+	"fmt"
+	"log"
+
+	"strings"
+
+	"time"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/pkg/errors"
+)
+
+func resourceCloudFlareZoneSettingsOverride() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudFlareZoneSettingsOverrideCreate,
+		Read:   resourceCloudFlareZoneSettingsOverrideRead,
+		Update: resourceCloudFlareZoneSettingsOverrideUpdate,
+		Delete: resourceCloudFlareZoneSettingsOverrideDelete,
+
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"settings": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: resourceCloudFlareZoneSettingsSchema,
+				},
+			},
+
+			"initial_settings": {
+				Type:     schema.TypeList,
```

Comment:
Is the ordering of settings significant, or guaranteed to always be returned in the same order?

Reply:
Maybe `TypeMap` is more appropriate here? I'll defer to you

Reply:
I think this runs into the problem with structured map values in terraform's schema language

Reply:
The order doesn't matter per se

Reply:
If ordering doesn't matter, or more importantly if ordering will not be guaranteed, then we should use a `TypeSet` here.

Reply:
This is always restricted to a single item in the terraform config so ordering will not matter (one to one relationship with a sub-resource that holds all the settings). It looks more understandable to use a TypeList in this case and I thought this was also the convention from looking at AWS/Google providers ?